## Script to determinate the target genes for a transcript
## form narrowPeak file generated by MaCS2.

## Autor: Sara Cartan Moya and Jorge Dominguez Barragan - saracartanmoya@gmail.com or jodombar@gmail.com
## Date: November 2019

## Installing chipseeker annotation package of the right organism (Arabidpsosi thaliana is used in this example). arg will be a character
## list
## Create a function which turns narroPeak into a parameter.

args <- commandArgs(trailingOnly = T)

input.file.name<- args[[1]]
promoter.length<- as.numeric(args[[2]])
output.file.name<- args[[3]]

library(ChIPseeker)
library(TxDb.Athaliana.BioMart.plantsmart28)
txdb <- TxDb.Athaliana.BioMart.plantsmart28


## Reading peak file
prr5.peaks <- readPeakFile(peakfile = "input.file.name",
                           header=FALSE)

## Defining the area around TSS which is considered as promoter.
promoter <- getPromoters(TxDb=txdb, 
                         upstream=promoter.length, 
                         downstream=promoter.length)

## Peak annotation
prr5.peakAnno <- annotatePeak(peak = prr5.peaks, 
                             tssRegion=c(-1000, 1000),
                             TxDb=txdb)

plotAnnoPie(prr5.peakAnno)
plotDistToTSS(prr5.peakAnno,
              title="Distribution of genomic loci relative to TSS",
              ylab = "Genomic Loci (%) (5' -> 3')")

## Turning annotation into data frame
prr5.annotation <- as.data.frame(prr5.peakAnno)
head(prr5.annotation)

target.genes <- prr5.annotation$geneId[prr5.annotation$annotation == "Promoter"]

write(x = target.genes,file = "experiment_target_genes.txt")

